src/idiomatic/leveraging-the-type-system/borrow-checker-invariants/single-use-values.md

```diff
@@ -0,0 +1,49 @@
+---
+minutes: 0
+---
+
+# Single-use values
+
+In some circumstances we want values that can be used _exactly once_. One critical example of this is in cryptography: "Nonces."
+
+```rust,editable
+fn main() {
+pub struct Key;
+
+// Pretend this is a cryptographically unique, use-once number.
+pub struct Nonce(u32);
+
+// It's unsafe to declare a nonce directly! In practice,
+// this would be done with an RNG source, and potentially
+// a timestamp.
+unsafe fn new_nonce_from_raw(nonce: u32) -> Nonce {
+    Nonce(nonce)
+}
+
+let nonce = unsafe { new_nonce_from_raw(1337) };
+let data_1: [u8; 4] = [1, 2, 3, 4];
+let data_2: [u8; 4] = [4, 3, 2, 1];
+let key = Key;
+
+// The key and data can be re-used, copied, etc. but the nonce cannot.
+fn encrypt(nonce: Nonce, key: &Key, data: &[u8]) {}
+
+encrypt(nonce, &key, &data_1);
+encrypt(nonce, &key, &data_2); // 🛠️❌
+}
+```
```

Comment:
I think the `new_non_from_raw`/`unsafe` aspect add unnecessary noise to the example. We don't need to go into the details as to how a `Nonce` is constructed, we just need a constructor/generator to exist.